api/hasura/cron/epochs.ts

```diff
@@ -273,78 +279,79 @@ export async function notifyEpochStart({
     }
 
     assert(circle, 'panic: no circle for epoch');
-    const epochStartDate = DateTime.fromISO(start_date);
-    const epochEndDate = DateTime.fromISO(end_date);
-    const eligibleUsersCount = circle.users_aggregate.aggregate?.count;
-
-    const message = dedent`
-      A new ${circle.organization?.name}/${circle.name} epoch is active!
-      ${eligibleUsersCount} users will be participating and the duration of the epoch will be:
-      **${epochStartDate.toLocaleString(
-        DateTime.DATETIME_FULL
-      )}** to **${epochEndDate.toLocaleString(DateTime.DATETIME_FULL)}**
-    `;
-
-    const {
-      discord_channel_id: channelId,
-      discord_role_id: roleId,
-      alerts,
-    } = circle?.discord_circle || {};
-
-    if (channelId && roleId && alerts?.['epoch-start']) {
-      await sendSocialMessage({
-        message,
-        circleId: circle.id,
-        channels: {
-          isDiscordBot: true,
-          discordBot: {
-            type: 'start' as const,
-            channelId,
-            roleId,
-            epochName: `Epoch ${epochNumber}`,
-            circleId: circle.id,
-            circleName: `${circle.organization?.name}/${circle.name}`,
-            startTime: start_date,
-            endTime: end_date,
-          },
-        },
-        sanitize: false,
-      });
-    }
-
-    if (circle.discord_webhook) {
-      await notifyEpochStatus(message, { discord: true }, epoch);
-    }
-
-    if (circle.telegram_id)
-      await notifyEpochStatus(message, { telegram: true }, epoch);
-
     if (epoch.circle) {
       await insertEpochStartActivity({
         ...epoch,
         organization_id: epoch.circle.organization.id,
       });
     }
 
-    const membersData = (epoch.circle?.users || []).reduce<{ email: string }[]>(
-      (acc, u) => {
+    if (!epoch.circle?.organization.sample) {
```

Comment:
just to be safe change this to `if(epoch.circle && !epoch.circle.organization.sample)` otherwise the circle being nil could trigger this

Reply:
This is in two places at least

Reply:
@CryptoGraffe  I changed to be !circle.organization.sample because circle is destructured and asserted few line before my line so it can't be null